Homekit test race condiation

**The problem**

Unrelated homekit test failures: https://github.com/home-assistant/core/pull/46422/checks?check_run_id=1888691090#step:8:169


**What is version of Home Assistant Core has the issue?**

core-dev

**What was the last working version of Home Assistant Core?**

_No response_

**What type of installation are you running?**

Home Assistant OS

**Integration causing the issue**

homekit

**Link to integration documentation on our website**

_No response_

**Example YAML snippet**

_No response_

**Anything in the logs that might be useful for us?**

```txt
_________________________ test_setup_imported[pyloop] __________________________
[gw1] linux -- Python 3.8.2 /__w/core/core/venv/bin/python

hass = <homeassistant.core.HomeAssistant object at 0x7f417f246280>
mock_zeroconf = <MagicMock name='HaZeroconf()' id='139919285421104'>

    async def test_setup_imported(hass, mock_zeroconf):
        """Test async_setup with imported config options."""
        legacy_persist_file_path = hass.config.path(HOMEKIT_FILE)
        legacy_aid_storage_path = hass.config.path(STORAGE_DIR, "homekit.aids")
        legacy_homekit_state_contents = {"homekit.state": 1}
        legacy_homekit_aids_contents = {"homekit.aids": 1}
        await hass.async_add_executor_job(
            _write_data, legacy_persist_file_path, legacy_homekit_state_contents
        )
        await hass.async_add_executor_job(
            _write_data, legacy_aid_storage_path, legacy_homekit_aids_contents
        )
    
        entry = MockConfigEntry(
            domain=DOMAIN,
            source=SOURCE_IMPORT,
            data={CONF_NAME: BRIDGE_NAME, CONF_PORT: DEFAULT_PORT, CONF_ENTRY_INDEX: 0},
            options={},
        )
        entry.add_to_hass(hass)
    
        with patch(f"{PATH_HOMEKIT}.HomeKit") as mock_homekit:
            mock_homekit.return_value = homekit = Mock()
            type(homekit).async_start = AsyncMock()
            assert await hass.config_entries.async_setup(entry.entry_id)
            await hass.async_block_till_done()
    
        mock_homekit.assert_any_call(
            hass,
            BRIDGE_NAME,
            DEFAULT_PORT,
            None,
            ANY,
            {},
            HOMEKIT_MODE_BRIDGE,
            None,
            entry.entry_id,
        )
        assert mock_homekit().setup.called is True
    
        # Test auto start enabled
        mock_homekit.reset_mock()
        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
        await hass.async_block_till_done()
    
        mock_homekit().async_start.assert_called()
    
        migrated_persist_file_path = get_persist_fullpath_for_entry_id(hass, entry.entry_id)
>       assert (
            await hass.async_add_executor_job(
                json_util.load_json, migrated_persist_file_path
            )
            == legacy_homekit_state_contents
        )
E       AssertionError: assert {} == {'homekit.state': 1}
E         Right contains 1 more item:
E         {'homekit.state': 1}
E         Full diff:
E         - {'homekit.state': 1}
E         + {}
```

--- a/homeassistant/components/homekit/__init__.py
+++ b/homeassistant/components/homekit/__init__.py
@@ -240,7 +240,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     if CONF_ENTRY_INDEX in conf and conf[CONF_ENTRY_INDEX] == 0:
         _LOGGER.debug("Migrating legacy HomeKit data for %s", name)
-        hass.async_add_executor_job(
+        await hass.async_add_executor_job(
             migrate_filesystem_state_data_for_primary_imported_entry_id,
             hass,
             entry.entry_id,

--- a/tests/components/homekit/test_homekit.py
+++ b/tests/components/homekit/test_homekit.py
@@ -51,7 +51,7 @@ from homeassistant.const import (
     CONF_PORT,
     DEVICE_CLASS_BATTERY,
     DEVICE_CLASS_HUMIDITY,
-    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STARTED,
     EVENT_HOMEASSISTANT_STOP,
     PERCENTAGE,
     SERVICE_RELOAD,
@@ -118,7 +118,7 @@ async def test_setup_min(hass, mock_zeroconf):
 
     # Test auto start enabled
     mock_homekit.reset_mock()
-    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
     await hass.async_block_till_done()
 
     mock_homekit().async_start.assert_called()
@@ -155,7 +155,7 @@ async def test_setup_auto_start_disabled(hass, mock_zeroconf):
     # Test auto_start disabled
     homekit.reset_mock()
     homekit.async_start.reset_mock()
-    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
     await hass.async_block_till_done()
     assert homekit.async_start.called is False
 
@@ -951,7 +951,7 @@ async def test_setup_imported(hass, mock_zeroconf):
 
     # Test auto start enabled
     mock_homekit.reset_mock()
-    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
     await hass.async_block_till_done()
 
     mock_homekit().async_start.assert_called()
@@ -1005,7 +1005,7 @@ async def test_yaml_updates_update_config_entry_for_name(hass, mock_zeroconf):
 
     # Test auto start enabled
     mock_homekit.reset_mock()
-    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
     await hass.async_block_till_done()
 
     mock_homekit().async_start.assert_called()

